<Type Name="StackOverflowException" FullName="System.StackOverflowException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="991aa9f4b94575c2f58703680ad95ac0d43e46c1" /><Meta Name="ms.sourcegitcommit" Value="9cee676b7864fa9dcb91319dcf5c339f3a724fbf" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="es-ES" /><Meta Name="ms.lasthandoff" Value="02/07/2019" /><Meta Name="ms.locfileid" Value="55828892" /></Metadata><TypeSignature Language="C#" Value="public sealed class StackOverflowException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable sealed beforefieldinit StackOverflowException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.StackOverflowException" />
  <TypeSignature Language="VB.NET" Value="Public NotInheritable Class StackOverflowException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class StackOverflowException sealed : SystemException" />
  <TypeSignature Language="F#" Value="type StackOverflowException = class&#xA;    inherit SystemException" />
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netframework-4.8">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
    <Attribute FrameworkAlternate="netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0">
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="cf64c-101">Excepción que se produce cuando la pila de ejecución se desborda debido a que contiene demasiadas llamadas a métodos anidadas.</span><span class="sxs-lookup"><span data-stu-id="cf64c-101">The exception that is thrown when the execution stack overflows because it contains too many nested method calls.</span></span> <span data-ttu-id="cf64c-102">Esta clase no puede heredarse.</span><span class="sxs-lookup"><span data-stu-id="cf64c-102">This class cannot be inherited.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="cf64c-103">`StackOverflowException` se produce para los errores de desbordamiento de pila de ejecución, normalmente en el caso de una recursividad muy profunda o ilimitada.</span><span class="sxs-lookup"><span data-stu-id="cf64c-103">`StackOverflowException` is thrown for execution stack overflow errors, typically in case of a very deep or unbounded recursion.</span></span> <span data-ttu-id="cf64c-104">Por lo tanto, asegúrese de que el código no tiene un bucle infinito o recursividad infinita.</span><span class="sxs-lookup"><span data-stu-id="cf64c-104">So make sure your code doesn't have an infinite loop or infinite recursion.</span></span>
  
 <span data-ttu-id="cf64c-105">`StackOverflowException` utiliza HRESULT COR_E_STACKOVERFLOW, que tiene el valor 0x800703E9.</span><span class="sxs-lookup"><span data-stu-id="cf64c-105">`StackOverflowException` uses the HRESULT COR_E_STACKOVERFLOW, which has the value 0x800703E9.</span></span> <span data-ttu-id="cf64c-106">El <xref:System.Reflection.Emit.OpCodes.Localloc> produce la instrucción del lenguaje intermedio (IL) `StackOverflowException`.</span><span class="sxs-lookup"><span data-stu-id="cf64c-106">The <xref:System.Reflection.Emit.OpCodes.Localloc> intermediate language (IL) instruction throws `StackOverflowException`.</span></span> <span data-ttu-id="cf64c-107">Para obtener una lista de valores de propiedad iniciales para un `StackOverflowException` de objetos, consulte el <xref:System.StackOverflowException.%23ctor%2A> constructores.</span><span class="sxs-lookup"><span data-stu-id="cf64c-107">For a list of initial property values for a `StackOverflowException` object, see the <xref:System.StackOverflowException.%23ctor%2A> constructors.</span></span>  
  
 <span data-ttu-id="cf64c-108">A partir de .NET Framework 2.0, no puede detectar un `StackOverflowException` objeto con un `try` / `catch` bloque y el proceso correspondiente finaliza de forma predeterminada.</span><span class="sxs-lookup"><span data-stu-id="cf64c-108">Starting with the .NET Framework 2.0, you can't catch a `StackOverflowException` object with a `try`/`catch` block, and the corresponding process is terminated by default.</span></span> <span data-ttu-id="cf64c-109">Por lo tanto, debe escribir el código para detectar y evitar un desbordamiento de pila.</span><span class="sxs-lookup"><span data-stu-id="cf64c-109">Consequently, you should write your code to detect and prevent a stack overflow.</span></span> <span data-ttu-id="cf64c-110">Por ejemplo, si la aplicación depende de la recursividad, utilice un contador o una condición de estado para finalizar el bucle recursivo.</span><span class="sxs-lookup"><span data-stu-id="cf64c-110">For example, if your app depends on recursion, use a counter or a state condition to terminate the recursive loop.</span></span>
<span data-ttu-id="cf64c-111">Consulte (Examples)(#examples) sección para ver una ilustración de esta técnica.</span><span class="sxs-lookup"><span data-stu-id="cf64c-111">See the (Examples)(#examples) section for an illustration of this technique.</span></span>
 
> [!NOTE]
>  <span data-ttu-id="cf64c-112">Aplicar el <xref:System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptionsAttribute> atributo a un método que produce una `StackOverflowException` no tiene ningún efecto.</span><span class="sxs-lookup"><span data-stu-id="cf64c-112">Applying the <xref:System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptionsAttribute> attribute to a method that throws a `StackOverflowException` has no effect.</span></span> <span data-ttu-id="cf64c-113">Todavía no se puede controlar la excepción del código de usuario.</span><span class="sxs-lookup"><span data-stu-id="cf64c-113">You still cannot handle the exception from user code.</span></span>

 <span data-ttu-id="cf64c-114">Si la aplicación hospeda common language runtime (CLR), puede especificar que el CLR debe descargar el dominio de aplicación cuando se produce una excepción de desbordamiento de pila y el proceso correspondiente pueda continuar.</span><span class="sxs-lookup"><span data-stu-id="cf64c-114">If your app hosts the common language runtime (CLR), it can specify that the CLR should unload the application domain where the stack overflow exception occurs and let the corresponding process continue.</span></span> <span data-ttu-id="cf64c-115">Para obtener más información, consulte [ICLRPolicyManager (interfaz)](~/docs/framework/unmanaged-api/hosting/iclrpolicymanager-interface.md).</span><span class="sxs-lookup"><span data-stu-id="cf64c-115">For more information, see [ICLRPolicyManager Interface](~/docs/framework/unmanaged-api/hosting/iclrpolicymanager-interface.md).</span></span>

## Examples

 <span data-ttu-id="cf64c-116">En el ejemplo siguiente se usa un contador para asegurarse de que el número de recursiva las llamadas a la `Execute` método no superar un máximo definido por la constante MAX_RECURSIVE_CALLS.</span><span class="sxs-lookup"><span data-stu-id="cf64c-116">The following example uses a counter to ensure that the number of recursive calls to the `Execute` method do not exceed a maximum defined by the MAX_RECURSIVE_CALLS constant.</span></span>
  
 [!code-csharp[System.StackOverflowException.Class#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.stackoverflowexception.class/cs/example1a.cs#1)]
 [!code-vb[System.StackOverflowException.Class#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.stackoverflowexception.class/vb/example1a.vb#1)]  

 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
    <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="cf64c-117">Controlar y generar excepciones</span><span class="sxs-lookup"><span data-stu-id="cf64c-117">Handling and Throwing Exceptions</span></span></related>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="cf64c-118">Inicializa una nueva instancia de la clase <see cref="T:System.StackOverflowException" />.</span><span class="sxs-lookup"><span data-stu-id="cf64c-118">Initializes a new instance of the <see cref="T:System.StackOverflowException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public StackOverflowException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.StackOverflowException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; StackOverflowException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="cf64c-119">Inicializa una nueva instancia de la clase <see cref="T:System.StackOverflowException" />, estableciendo la propiedad <see cref="P:System.Exception.Message" /> de una nueva instancia en un mensaje proporcionado por el sistema que describe el error, como "La operación solicitada provocó un desbordamiento de la pila".</span><span class="sxs-lookup"><span data-stu-id="cf64c-119">Initializes a new instance of the <see cref="T:System.StackOverflowException" /> class, setting the <see cref="P:System.Exception.Message" /> property of the new instance to a system-supplied message that describes the error, such as "The requested operation caused a stack overflow."</span></span> <span data-ttu-id="cf64c-120">Este mensaje tiene en cuenta la referencia cultural del sistema actual.</span><span class="sxs-lookup"><span data-stu-id="cf64c-120">This message takes into account the current system culture.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="cf64c-121">En la tabla siguiente se muestran los valores de propiedad iniciales para una instancia de <xref:System.StackOverflowException>.</span><span class="sxs-lookup"><span data-stu-id="cf64c-121">The following table shows the initial property values for an instance of <xref:System.StackOverflowException>.</span></span>  
  
|<span data-ttu-id="cf64c-122">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="cf64c-122">Property</span></span>|<span data-ttu-id="cf64c-123">Valor</span><span class="sxs-lookup"><span data-stu-id="cf64c-123">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="cf64c-124">`null`.</span><span class="sxs-lookup"><span data-stu-id="cf64c-124">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="cf64c-125">La cadena del mensaje de error localizado.</span><span class="sxs-lookup"><span data-stu-id="cf64c-125">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public StackOverflowException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.StackOverflowException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; StackOverflowException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new StackOverflowException : string -&gt; StackOverflowException" Usage="new System.StackOverflowException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="cf64c-126">Objeto <see cref="T:System.String" /> que describe el error.</span><span class="sxs-lookup"><span data-stu-id="cf64c-126">A <see cref="T:System.String" /> that describes the error.</span></span> <span data-ttu-id="cf64c-127">Se pretende que el contenido de message sea inteligible.</span><span class="sxs-lookup"><span data-stu-id="cf64c-127">The content of message is intended to be understood by humans.</span></span> <span data-ttu-id="cf64c-128">El autor de llamada de este constructor debe asegurarse de que la cadena se haya adaptado para la referencia cultural actual del sistema.</span><span class="sxs-lookup"><span data-stu-id="cf64c-128">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span></span></param>
        <summary><span data-ttu-id="cf64c-129">Inicializa una nueva instancia de la clase <see cref="T:System.StackOverflowException" /> con el mensaje de error especificado.</span><span class="sxs-lookup"><span data-stu-id="cf64c-129">Initializes a new instance of the <see cref="T:System.StackOverflowException" /> class with a specified error message.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="cf64c-130">En la tabla siguiente se muestran los valores de propiedad iniciales para una instancia de <xref:System.StackOverflowException>.</span><span class="sxs-lookup"><span data-stu-id="cf64c-130">The following table shows the initial property values for an instance of <xref:System.StackOverflowException>.</span></span>  
  
|<span data-ttu-id="cf64c-131">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="cf64c-131">Property</span></span>|<span data-ttu-id="cf64c-132">Valor</span><span class="sxs-lookup"><span data-stu-id="cf64c-132">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="cf64c-133">Referencia nula (`Nothing` en Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="cf64c-133">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="cf64c-134">Cadena con el mensaje de error.</span><span class="sxs-lookup"><span data-stu-id="cf64c-134">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public StackOverflowException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.StackOverflowException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; StackOverflowException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new StackOverflowException : string * Exception -&gt; StackOverflowException" Usage="new System.StackOverflowException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="cf64c-135">Mensaje de error que explica el motivo de la excepción.</span><span class="sxs-lookup"><span data-stu-id="cf64c-135">The error message that explains the reason for the exception.</span></span></param>
        <param name="innerException"><span data-ttu-id="cf64c-136">Excepción que es la causa de la excepción actual.</span><span class="sxs-lookup"><span data-stu-id="cf64c-136">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="cf64c-137">Si el parámetro <paramref name="innerException" /> no es una referencia nula (<see langword="Nothing" /> en Visual Basic), la excepción actual se inicia en un bloque <see langword="catch" /> que controla la excepción interna.</span><span class="sxs-lookup"><span data-stu-id="cf64c-137">If the <paramref name="innerException" /> parameter is not a null reference (<see langword="Nothing" /> in Visual Basic), the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="cf64c-138">Inicializa una nueva instancia de la clase <see cref="T:System.StackOverflowException" /> con el mensaje de error especificado y una referencia a la excepción interna que representa la causa de esta excepción.</span><span class="sxs-lookup"><span data-stu-id="cf64c-138">Initializes a new instance of the <see cref="T:System.StackOverflowException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="cf64c-139">Una excepción que se produce como resultado directo de una excepción anterior puede incluir una referencia a la excepción anterior de la propiedad <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="cf64c-139">An exception that is thrown as a direct result of a previous exception can include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="cf64c-140">La propiedad <xref:System.Exception.InnerException%2A> devuelve el mismo valor pasado al constructor, o una referencia nula (`Nothing` en Visual Basic) si la propiedad <xref:System.Exception.InnerException%2A> no proporciona el valor de la excepción interna al constructor.</span><span class="sxs-lookup"><span data-stu-id="cf64c-140">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or a null reference (`Nothing` in Visual Basic) if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="cf64c-141">En la tabla siguiente se muestran los valores de propiedad iniciales para una instancia de <xref:System.StackOverflowException>.</span><span class="sxs-lookup"><span data-stu-id="cf64c-141">The following table shows the initial property values for an instance of <xref:System.StackOverflowException>.</span></span>  
  
|<span data-ttu-id="cf64c-142">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="cf64c-142">Property</span></span>|<span data-ttu-id="cf64c-143">Valor</span><span class="sxs-lookup"><span data-stu-id="cf64c-143">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="cf64c-144">Referencia a la excepción interna.</span><span class="sxs-lookup"><span data-stu-id="cf64c-144">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="cf64c-145">Cadena con el mensaje de error.</span><span class="sxs-lookup"><span data-stu-id="cf64c-145">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="cf64c-146">Controlar y generar excepciones</span><span class="sxs-lookup"><span data-stu-id="cf64c-146">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
  </Members>
</Type>
